packaging/rpm/flightctl.spec

```diff
@@ -1,74 +1,127 @@
+# Disable debug information package creation
 %define debug_package %{nil}
 
-Name: flightctl
-Version: 0.3.0
-Release: 1%{?dist}
-Summary: Flightctl CLI
+# Define the Go Import Path
+%global goipath github.com/flightctl/flightctl
+Version:    0.3.0
+%gometa
+
+Name:       flightctl
+Release:    1%{?dist}
+Summary:    Flightctl is a manager of the edge device fleets.
 
-License: XXX
-URL: https://github.com/flightctl/flightctl
-Source0: flightctl-0.3.0.tar.gz
+License:    Apache-2.0 AND BSD-2-Clause AND BSD-3-Clause AND ISC AND MIT
+URL:        %{gourl}
+Source0:    %{gosource}
+Source1:    %{archivename}-vendor.tar.bz2
 
-BuildRequires: golang
-BuildRequires: make
+BuildRequires: go-rpm-macros
 BuildRequires: git
+BuildRequires: compiler(go-compiler)
 BuildRequires: openssl-devel
+
 Requires: openssl
 
+# Skip description for the main package since it won't be created
 %description
-Flightctl is a command line interface for managing edge device fleets.
+# Main package is empty and not created.
 
+# cli sub-package
+%package cli
+Summary: Flightctl CLI
+%description cli
+Flightctl is a command line interface for managing edge device fleets.
 
+# agent sub-package
 %package agent
 Summary: Flightctl Agent
 %description agent
 Flightctl Agent is a component of the flightctl tool.
 
 %prep
+%goprep -A
 %setup -q -n flightctl-0.0.1
+./packaging/rpm/get-git-sources.sh
+./packaging/rpm/get-vendored-sources.sh
 
 %build
-
 # if this is a buggy version of go we need to set GOPROXY as workaround
 # see https://github.com/golang/go/issues/61928
 GOENVFILE=$(go env GOROOT)/go.env
 if [[ ! -f "{$GOENVFILE}" ]]; then
     export GOPROXY='https://proxy.golang.org,direct'
 fi
-make build
+make build-cli build-agent \
+SOURCE_GIT_TAG=$(git describe --tags --exclude latest) \
```

Comment:
Would it make sense to merge all these into the Makefile, so we have a central place to compute them?

Reply:
It is inside, I'm not sure (yet) why it doesn't work there...